### PR TITLE
Chore(cx 1421): replace ios native analytics with the rn segment analytics

### PR DESCRIPTION
**Pull request by lordkiz**
The type of this PR is: **TYPE**

This PR resolves [CX-1421]

Snapshot tests fixed by: @brainbicycle 🙌🏾
### Description
The ticket necessitated the following changes:
- Removes ARAnalytics. All events and screen tracking events are bubbled up to react native from where we use react-native-segment to log these.
- Removes Sailtthru
- Removes ARAnalytics visualiser and AR Analytics integrations
- Minor initial AppBoy config
- Migrates SDWebImage from v3.x to v5.8.3 (builds off the work by @gkartalis here: https://github.com/artsy/eigen/pull/5021)

### Follow Up Tasks:
- Verify all events are arriving at segment
- Verify Appboy basics such as sessions are logged on braze

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a CHANGELOG.yml entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Replace ios native analytics with segment analytics

</details>

[CX-1421]: https://artsyproduct.atlassian.net/browse/CX-1421

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7e4161264ba08da6b87e7881ef46ffcc63e4f231
</p>

File: HACKS.md [hunk 1]
```diff
@@ -198,18 +198,6 @@ As you can see in the PR and issue, android doesn't use ellipsis on the placehol
...
-#### When can we remove this:
-
-Once we work on [CX-1421](https://artsyproduct.atlassian.net/browse/CX-1421?atlOrigin=eyJpIjoiYjJiMWY4MThlOTY2NDc3ZGJiMTljZTJjM2FmMmVkZjQiLCJwIjoiaiJ9) to replace the native ios analytics with the TS analytics we use for android already.
-
-#### Explanation/Context:
-
-Basically these patches remove the podspecs of these two deps. They confuse and break ios, and we don't use them there yet anyway.
-
-Once we start working on replacing the native analytics with TS ones, then we remove these two patches too.
-
```
Comment (brainbicycle) [inline HACKS.md:212]: 🎉

File: Artsy/App/ARAppDelegate.m [hunk 2]
```diff
@@ -229,11 +226,6 @@ - (void)forceCacheCustomFonts
 
 - (void)finishOnboarding:(AROnboardingViewController *)viewController animated:(BOOL)animated
 {
-    // We now have a proper Artsy user, not just a local temporary ID
-    // So we have to re-identify the analytics user
-    // to ensure we start sending the Gravity ID as well as the local temporary ID
-    [ARUserManager identifyAnalyticsUser];
-
```
Comment (brainbicycle) [inline Artsy/App/ARAppDelegate.m:236]: We may have to do Braze changeUser call here, unless that is something that can be handled on RN side: https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/analytics/setting_user_ids/#suggested-user-id-naming-convention

Comment (brainbicycle) [inline Artsy/App/ARAppDelegate.m:236]: We may have to do Braze changeUser call here, unless that is something that can be handled on RN side: https://www.braze.com/docs/developer_guide/platform_integration_guides/ios/analytics/setting_user_ids/#suggested-user-id-naming-convention, can we add a TODO similar to the other methods?

File: Artsy/App/ARAppActivityContinuationDelegate.m [hunk 3]
```diff
@@ -45,11 +44,13 @@ - (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserAct
...
-    if ([URL.host isEqualToString:SailthruLinkDomain]) {
-        [ArtsyAPI getDecodedURLAndRegisterClick:URL completion:callback];
-    } else {
-        callback(URL);
-    }
+    // if ([URL.host isEqualToString:SailthruLinkDomain]) {
+    //     [ArtsyAPI getDecodedURLAndRegisterClick:URL completion:callback];
+    // } else {
+    //     callback(URL);
+    // }
```
Comment (brainbicycle) [inline Artsy/App/ARAppActivityContinuationDelegate.m:51]:
```suggestion
```

Can we remove the commented out code here!

File: Artsy/App/ARAppDelegate+Analytics.m [hunk 4]
```diff
@@ -1,103 +1,20 @@
...
-        case UIUserInterfaceStyleUnspecified:
-            [ARAnalytics setUserProperty:@"user interface style" toValue:@"unspecified"];
-            break;
-        case UIUserInterfaceStyleLight:
-            [ARAnalytics setUserProperty:@"user interface style" toValue:@"light"];
-            break;
-        case UIUserInterfaceStyleDark:
-            [ARAnalytics setUserProperty:@"user interface style" toValue:@"dark"];
-            break;
-    }
```
Comment (brainbicycle) [inline Artsy/App/ARAppDelegate+Analytics.m:100]: If we can we should still track these events + properties using your new analytics method!

File: Artsy/App/ARAppDelegate+Analytics.m [hunk 5]
```diff
@@ -1,103 +1,20 @@
...
-
-    // For OSS builds don't ship the sentry env
-    if (sentryEnv && ![sentryEnv isEqualToString:@"-"]) {
-        id sentry = [[ARSentryAnalyticsProvider alloc] initWithDSN:sentryEnv];
-        [ARAnalytics setupProvider:sentry];
-    }
-
-    if ([AROptions boolForOption:AROptionsShowAnalyticsOnScreen]) {
-        ARAnalyticsVisualizer *visualizer = [ARAnalyticsVisualizer new];
-        [ARAnalytics setupProvider:visualizer];
```
Comment (brainbicycle) [inline Artsy/App/ARAppDelegate+Analytics.m:82]: The analytics visualizer is something @mikehrom uses pretty frequently to debug analytics code, we probably want to get this capability back pretty soon but I don't want to hold up this PR which is already a herculean effort! 

I'll create a ticket to make a cross platform version.

File: Artsy/App/ARAppDelegate.m [hunk 6]
```diff
@@ -174,9 +168,10 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
     [self setupForAppLaunch];
 
     NSString *brazeAppKey = [ReactNativeConfig envFor:@"BRAZE_PRODUCTION_APP_KEY_IOS"];
+    NSString *brazeSDKEndPoint = @"sdk.iad-06.braze.com";
 
     NSMutableDictionary *appboyOptions = [NSMutableDictionary dictionary];
-    appboyOptions[ABKAppboyEndpointDelegateKey] = self;
+    appboyOptions[ABKEndpointKey] = brazeSDKEndPoint;
```
Comment (brainbicycle) [inline Artsy/App/ARAppDelegate.m:174]: nice ✨

File: Artsy/App/ARAppDelegate+Analytics.m (hunk 4 above)
Comment (lordkiz) [reply Artsy/App/ARAppDelegate+Analytics.m:100]: This is already tracked in react native. We populate the traits with react-native's `Appearance.scheme`

File: Artsy/App/ARAppActivityContinuationDelegate.m (hunk 3 above)
Comment (lordkiz) [reply Artsy/App/ARAppActivityContinuationDelegate.m:51]: I forgot this. Will remove.

File: Artsy/App/ARAppDelegate.m (hunk 2 above)
Comment (lordkiz) [reply Artsy/App/ARAppDelegate.m:236]: @brainbicycle I thought by using Braze segment integration, identify calls automatically are forwarded to the providers. My experience with firebase, adjust etc segment's integration is that these are auto forwarded to the correct provider. For example, calling Segment.identify() would do identify related stuffs on its integrations.

File: Artsy/App/ARAppDelegate+Analytics.m (hunk 5 above)
Comment (mikehrom) [reply Artsy/App/ARAppDelegate+Analytics.m:82]: Ok thank you - we would be a little lost without the visualizer 😅

File: Artsy/App/ARAppDelegate.m (hunk 2 above)
Comment (brainbicycle) [reply Artsy/App/ARAppDelegate.m:236]: Makes sense, and you are probably right, let's leave it and see if it is a problem.

File: Artsy/App/ARAppDelegate.m (hunk 6 above)
Comment (brainbicycle) [reply Artsy/App/ARAppDelegate.m:174]: Actually realizing if we no longer need the delegate for this braze version we can get rid of this method: 
https://github.com/artsy/eigen/blob/f53ee86b46ec38febc7d7ad8cbb8eedaccc971d3/Artsy/App/ARAppDelegate.m#L188

File: Artsy/App/ARAppDelegate+Analytics.m (hunk 5 above)
Comment (brainbicycle) [reply Artsy/App/ARAppDelegate+Analytics.m:82]: ticket here, this could be a good knowledge share task: https://artsyproduct.atlassian.net/browse/CX-1721

**Comment by brainbicycle** [comment]
Awesome work! I'm excited to get this in, since it is such a big change I would prefer if we get it in sooner and we should probably do some analytics specific QA ahead of next weeks release. Thanks for all the effort getting this over the line!

File: Artsy/View_Controllers/Auction/RefinementOptionsViewController.swift [hunk 7]
```diff
@@ -128,10 +128,10 @@ class RefinementAnalyticsOption: NSObject {
     }
 
     func sendAsEvent() {
-        ARAnalytics.event(name, withProperties: properties)
+        AREmission.sharedInstance().sendEvent(name, traits: properties)
```
Comment (pvinis) [inline Artsy/View_Controllers/Auction/RefinementOptionsViewController.swift:131]: I dont like that this is in emission and not in ArtsyNativeModule, but I guess we can move it later when we start removing things from emission.

File: Podfile [hunk 8]
```diff
@@ -82,7 +82,7 @@ target 'Artsy' do
   pod 'AFNetworking', '~> 2.5', subspecs: %w[Reachability Serialization Security NSURLSession NSURLConnection]
   pod 'AFOAuth1Client', git: 'https://github.com/artsy/AFOAuth1Client.git', tag: '0.4.0-subspec-fix'
   pod 'AFNetworkActivityLogger'
-  pod 'SDWebImage', '>= 3.7.2' # 3.7.2 contains a fix that allows you to not force decoding each image, which uses lots of memory
+  pod 'SDWebImage', '5.8.3'
```
Comment (pvinis) [inline Podfile:85]: NOICE!!!

File: Podfile [hunk 9]
```diff
@@ -129,11 +129,6 @@ target 'Artsy' do
   pod 'GoogleSignIn', '~> 5.0.2'
   pod 'Firebase/Auth'
 
-  # Analytics
-  pod 'Analytics'
-  pod 'ARAnalytics', subspecs: %w[Segmentio]
```
Comment (pvinis) [inline Podfile:134]: YESSSSSSS!!

File: Podfile [hunk 10]
```diff
@@ -129,11 +129,6 @@ target 'Artsy' do
...
-  pod 'Appboy-iOS-SDK'
```
Comment (pvinis) [inline Podfile:135]: Dont we need this? 🤔

File: HACKS.md [hunk 11]
```diff
@@ -198,18 +198,6 @@ As you can see in the PR and issue, android doesn't use ellipsis on the placehol
 
 We added a workaround on Input, to accept an array of placeholders, from longest to shortest, so that android can measure which one fits in the TextInput as placeholder, and it uses that. When android can handle a long placeholder and use ellipsis or if we don't use long placeholders anymore, this can go.
 
-# `@segment/analytics-react-native` and `@segment/analytics-react-native-appboy` patch
```
Comment (pvinis) [inline HACKS.md:201]: So good!

File: Artsy/View_Controllers/Util/Errors/ARNetworkErrorManager.m [hunk 12]
```diff
@@ -30,7 +30,7 @@ + (void)presentActiveError:(NSError *)error;
 
 + (void)presentActiveError:(NSError *)error withMessage:(NSString *)message;
 {
-    [ARAnalytics error:error withMessage:message];
+    [SentrySDK captureError:error];
```
Comment (pvinis) [inline Artsy/View_Controllers/Util/Errors/ARNetworkErrorManager.m:33]: Nice touch.

File: src/lib/AndroidApp.tsx [hunk 13]
```diff
@@ -20,7 +20,7 @@ import { Onboarding } from "./Scenes/Onboarding/Onboarding"
 import { ConsoleTrackingProvider } from "./utils/track/ConsoleTrackingProvider"
 import { AnalyticsConstants } from "./utils/track/constants"
 
-addTrackingProvider("segment rn android", SegmentTrackingProvider)
+addTrackingProvider(SEGMENT_TRACKING_PROVIDER, SegmentTrackingProvider)
```
Comment (pvinis) [inline src/lib/AndroidApp.tsx:23]: This name is more of a thing for us when we debug, not used anywhere else, so moving it there is not really doing anything.

File: src/lib/utils/track/SegmentTrackingProvider.ts [hunk 14]
```diff
@@ -61,7 +72,16 @@ export const SegmentTrackingProvider: TrackingProvider = {
...
+      analytics.track(event_name, rest as any)
+      return
+    }
+
     console.warn("oh wow, we are not tracking this event!! we should!", { info })
     assertNever(info)
   },
 }
+
+export default SegmentTrackingProvider
```
Comment (pvinis) [inline src/lib/utils/track/SegmentTrackingProvider.ts:87]: I think we should remove this. First, we don't import it as default anywhere. Second, we prefer to have named exports.

File: emission/Pod/Classes/Core/AREmission.m [hunk 15]
```diff
@@ -87,6 +87,25 @@ - (void)updateState:(NSDictionary *)state
...
+    NSMutableDictionary *screenTraits = [traits mutableCopy];
+    [screenTraits addEntriesFromDictionary:@{@"screen_name": screenName }];
+    [self.notificationsManagerModule sendEvent:screenTraits];
+}
+
+- (void)sendIdentifyEvent:(NSDictionary *)traits
+{
+    [self.notificationsManagerModule sendIdentifyEvent:traits];
+}
+
```
Comment (pvinis) [inline emission/Pod/Classes/Core/AREmission.m:108]: Did you end up testing that the pages/screens do send a screen track, and the others are just an event track? (Im guessing you did, but I just want to verify.)

File: Artsy/App/ARAppDelegate.h [hunk 16]
```diff
@@ -21,9 +20,6 @@
 @property (strong, nonatomic, readonly) NSString *referralURLRepresentation;
 @property (strong, nonatomic, readonly) NSString *landingURLRepresentation;
 
-/// Shared Sailthru instance.
-@property (strong, readonly) SailthruMobile *sailThru;
```
Comment (pvinis) [inline Artsy/App/ARAppDelegate.h:25]: See ya later sailthru :p

**Comment by pvinis** [comment]
Huuuuuuge props Kizito, very nice work. I added some comments, some celebrations, some questions. Sorry for the late response.

File: emission/Pod/Classes/Core/AREmission.m (hunk 15 above)
Comment (lordkiz) [reply emission/Pod/Classes/Core/AREmission.m:108]: Yes I did. These native screen events will contain `screen_name` in the traits bubbled up. And this is used as an identifier to know if this is intended as screen tracking. Native events will be bubbled up with `event_name`. The traits can never contain both.

File: src/lib/utils/track/SegmentTrackingProvider.ts (hunk 14 above)
Comment (lordkiz) [reply src/lib/utils/track/SegmentTrackingProvider.ts:87]: @pvinis I used it as a default in AuthModel.ts
I was really having trouble with tests when using the mocked named exported module with jest. I could only get it to work after mocking the default. I might have been missing something.

File: src/lib/AndroidApp.tsx (hunk 13 above)
Comment (lordkiz) [reply src/lib/AndroidApp.tsx:23]: I didn't know this. 😄

File: Podfile (hunk 10 above)
Comment (lordkiz) [reply Podfile:135]: We already get them from the segment integrations

File: Artsy/View_Controllers/Auction/RefinementOptionsViewController.swift (hunk 7 above)
Comment (lordkiz) [reply Artsy/View_Controllers/Auction/RefinementOptionsViewController.swift:131]: I thought so too, but I decided to use emission since we are already using it to bubble up STATE_CHANGED etc. I thought having it all in one place for now would make it easier to refactor later

File: src/lib/utils/track/SegmentTrackingProvider.ts (hunk 14 above)
Comment (pvinis) [reply src/lib/utils/track/SegmentTrackingProvider.ts:87]: I think I fixed it in the visualizer pr 👍
